resourseleak/TestConfidenceScores.java

```diff
@@ -0,0 +1,114 @@
+package resourceleak;
+
+import com.amazon.accessdeviceregisterservice.RegisterAVSDeviceResponse;
+import com.amazon.arest.client.aaa.AaaFeature;
+import com.amazon.coral.metrics.Metrics;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.a4b.endpointoperationservice.dependency.accessor.MediaStorageServiceAccessor;
+import com.amazonaws.a4b.endpointoperationservice.dependency.exception.DependencyServiceException;
+import com.amazonaws.a4b.endpointoperationservice.logic.OperationLogic;
+import com.amazonaws.a4b.endpointoperationservice.logic.S3Logic;
+import com.amazonaws.a4b.endpointoperationservice.model.OperationModel;
+import com.amazonaws.a4b.endpointoperationservice.stepfunctions.model.UploadMediaActivityInput;
+import com.amazonaws.a4b.endpointoperationservice.util.SerializationUtils;
+import com.amazonaws.a4b.platform.common.exception.HttpClientException;
+import com.amazonaws.a4b.platform.exception.InvalidDeviceStateException;
+import com.amazonaws.a4b.platform.utils.HttpClientHelper;
+import com.amazonaws.services.s3.model.S3Object;
+import org.apache.commons.io.IOUtils;
+
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.time.Clock;
+
+public class TestConfidenceScores {
+
+    private static final int CONNECTION_TIME_OUT = 3000;
+    private static final String APP_NAME = "A4BAnalyticsLambda";
+    private final MediaStorageServiceAccessor mssAccessor;
+    private final S3Logic s3Logic;
+    private final OperationLogic operationLogic;
+    private final Clock clock;
+
+
+    public String testConfidenceScores1() throws IOException {
+        final StringWriter writer = new StringWriter();
+        IOUtils.copy(this.getClass().getResourceAsStream("/report_template.html"), writer);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Currently, there are execution paths that do not contain closure statements, for example, when IOUtils.copy throws an exception. Either a) close the object returned by *getResourceAsStream()* in a try-finally block or b) close the resource by declaring the object returned by *getResourceAsStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).